src/main/java/com/epam/dataservice/HomeWork3.java

```diff
@@ -0,0 +1,154 @@
+package com.epam.dataservice;
+
+import com.epam.data.RoadAccident;
+import com.epam.data.RoadAccidentBuilder;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Created by bill on 16-5-1.
+ */
+public class HomeWork3 {
+    private final int MAX_READ_THREADS = 2;
+    private final int MAX_PROC_THREADS = 4;
+    public static final int MAX_ENRICH_THREADS = 100;
+    private final int MAX_WRITE_THREADS = 2;
+    public static final int MAX_BATCH_SIZE = 40000;
+    public static final String outputFileNames[] = {"NighttimeAccidents.csv", "DaytimeAccidents.csv"};
+
+    private LinkedList<String> fileList;
+    private BlockingQueue<List<RoadAccident>> readQueue;
+    private ExecutorService readExecutor;
+    private List<Future<Integer>> readTaskList = new ArrayList<>();
+
+    private ExecutorService procExecutor;
+
+    private List<BlockingQueue<RoadAccident>> writeQueues = new ArrayList<>(MAX_WRITE_THREADS);
+    private ExecutorService writeExecutor;
+
+    public HomeWork3(LinkedList<String> fileList) {
+        this.fileList = fileList;
+    }
+
+    public void startTask() {
+        readQueue = new ArrayBlockingQueue<List<RoadAccident>>(MAX_PROC_THREADS);
+        for (int i = 0; i < MAX_WRITE_THREADS; i++) {
+            writeQueues.add( new ArrayBlockingQueue<RoadAccident>(MAX_BATCH_SIZE/2) );
+        }
+
+        readExecutor = Executors.newFixedThreadPool(MAX_READ_THREADS);
+        for (String file : fileList ) {
+            FutureTask<Integer> readTask = new FutureTask<Integer>(
+                    new AccidentBatchLoader(MAX_BATCH_SIZE,readQueue,file));
+            readTaskList.add(readTask);
+            readExecutor.submit(readTask);
+        }
+
+        procExecutor = Executors.newFixedThreadPool(MAX_PROC_THREADS);
+        for (int i = 0; i < MAX_PROC_THREADS; i++) {
+            procExecutor.submit(new AccidentBatchProcessor(readQueue, writeQueues));
+        }
+
+        writeExecutor = Executors.newFixedThreadPool(MAX_WRITE_THREADS);
+        for (int i = 0; i < MAX_WRITE_THREADS; i++) {
+            writeExecutor.submit(new AccidentBatchWriter(writeQueues.get(i),outputFileNames[i]));
+        }
+        System.out.println("All task submitted");
+    }
+
+    public void run() {
+        startTask();
+
+        try {
+            readQueueFinish();
+            endDatLoading(procExecutor);
+            for (int i = 0; i < MAX_WRITE_THREADS; i++) {
+                System.out.println("Send quit message to writeQueue, count: " + Integer.toString(i+1));
+                writeQueues.get(i).put( new RoadAccidentBuilder(null).build() );
+            }
+            endDatLoading(writeExecutor);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void readQueueFinish() throws Exception {
```

Comment:
I see what you are trying to do here, but I believe there should be a better way to do it.... 
Don't know it yet though :) 
Let's discuss this code tomorrow.